src/libstore/build/local-derivation-goal.cc

```diff
@@ -359,7 +359,7 @@ bool LocalDerivationGoal::cleanupDecideWhetherDiskFull()
             if (buildMode != bmCheck && status.known->isValid()) continue;
             auto p = worker.store.printStorePath(status.known->path);
             if (pathExists(chrootRootDir + p))
-                renameFile((chrootRootDir + p), p);
+                moveFile((chrootRootDir + p), p);
```

Comment:
Do you have an idea if we should run our overlay using redirect_dir? Or should NixOS be careful about the mnt differences while doing the renames?

Reply:
s/NixOS/nix

Reply:
Nix should definitely be more careful and not assume too eagerly that a `rename` is possible. Now using `redirect_dir` might be worth it too, both to make sure that it still works with older Nix versions and because the fallback path is noticeably slower. The only thing to be careful about is that it seems to make overlayFS a bit more strict when it comes to changing the underlying directory (see [that section in particular](https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html?highlight=overlayfs#changes-to-underlying-filesystems)), but I don't think it would matter for sharing a Nix store.